pkg/R/pairRDD.R

```diff
@@ -742,7 +651,8 @@ setMethod("fullOuterJoin",
               joinTaggedList(v, list(TRUE, TRUE))
             }
 
-            joined <- flatMapValues(groupByKey(unionRDD(xTagged, yTagged), numPartitions), doJoin)
+            joined <- flatMapValues(groupByKey(unionRDD(xTagged, yTagged), numToInt(numPartitions)),
```

Comment:
while this change to using `numToInt` is a good one, lets not do it in the merge PR. we can do it in a follow up PR